Handler/Cancel.hs

```diff
@@ -0,0 +1,22 @@
+module Handler.Cancel where
+
+import Import
+import Stripe
+
+postCancelR :: Handler Html
+postCancelR = do
+    plan <- runDB $ getBy404 $ UniquePlan freePlanId
+    Entity userId user <- requireAuth
+
+    result <- runStripe $ cancelSubscription $ userStripeId user
+
+    case result of
+        Just _ -> do
+            runDB $ update userId [UserPlan =. entityKey plan]
+
+            setMessage "Plan changed!"
+            redirect RootR
+
+        _ -> do
```

Comment:
It's common to just use a catch-all wildcard even in cases where all but one pattern has already been matched.